webapp custom generator should not require to set a CMD configuration

### Description
Setting up a webapp (war) project with this configuration in the `build.gradle` file
```
kubernetes {
    generator {
        config {
            'webapp' {
                from = "tomcat:jdk11-openjdk-slim"
                targetDir = "/usr/local/tomcat/webapps"
            }
        }
    }
}
```
is failling to build with the error
```
> Task :k8sBuild FAILED
k8s: Running in Kubernetes mode
k8s: Running generator webapp
k8s: webapp: Using tomcat:jdk11-openjdk-slim as base image for webapp

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':k8sBuild'.
> Argument conflict: either shell or args should be specified and only in one form.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':k8sBuild'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:188)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:186)
[...]
Caused by: java.lang.IllegalArgumentException: Argument conflict: either shell or args should be specified and only in one form.
        at org.eclipse.jkube.kit.config.image.build.Arguments.validate(Arguments.java:109)
        at org.eclipse.jkube.kit.config.image.build.BuildConfiguration.initAndValidate(BuildConfiguration.java:336)
        at org.eclipse.jkube.kit.build.service.docker.helper.ConfigHelper.initAndValidate(ConfigHelper.java:245)
        at org.eclipse.jkube.kit.build.service.docker.helper.ConfigHelper.initAndValidate(ConfigHelper.java:134)
        at org.eclipse.jkube.kit.build.service.docker.helper.ConfigHelper.initImageConfiguration(ConfigHelper.java:209)
[...]
BUILD FAILED in 814ms
1 actionable task: 1 executed
```

Adding `cmd` to the configuration is fixing it:
```
kubernetes {
    generator {
        config {
            'webapp' {
                from = "tomcat:jdk11-openjdk-slim"
                targetDir = "/usr/local/tomcat/webapps"
                cmd = "catalina.sh run"
            }
        }
    }
}
```

However, it should not fail without `cmd` and use the command that is provided by the container image.

### Info

* Eclipse JKube version : 1.6.0

* Kubernetes / Red Hat OpenShift setup and version :
Minikube

* Sample Reproducer Project
Using the `webapp` quickstart and adding the configuration to it.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,9 +21,10 @@ Usage:
 ./scripts/extract-changelog-for-version.sh 1.3.37 5
 ```
 ### 1.8.0-SNAPSHOT
-* Fix #1260: Add documentation for PodAnnotationEnricher
-* Fix #1295: Spring Boot actuator endpoints failed to generate automatically if `deployment.yml` resource fragment is used
 * Fix #1201: ThorntailV2Generator works with Gradle Plugins
+* Fix #1260: Add documentation for PodAnnotationEnricher
+* Fix #1284: webapp custom generator should not require to set a CMD configuration
+* Fix #1295: Spring Boot actuator endpoints failed to generate automatically if `deployment.yml` resource fragment is used
 
 ### 1.7.0 (2022-02-25)
 * Fix #1315: Pod Log Service works for Jobs with no selector

--- a/jkube-kit/generator/webapp/src/main/java/org/eclipse/jkube/generator/webapp/WebAppGenerator.java
+++ b/jkube-kit/generator/webapp/src/main/java/org/eclipse/jkube/generator/webapp/WebAppGenerator.java
@@ -108,8 +108,12 @@ public class WebAppGenerator extends BaseGenerator {
 
     buildBuilder.from(getFrom(handler))
         .ports(handler.exposedPorts())
-        .cmd(Arguments.builder().shell(getDockerRunCommand(handler)).build())
         .env(getEnv(handler));
+
+    String dockerRunCommand = getDockerRunCommand(handler);
+    if (dockerRunCommand != null) {
+      buildBuilder.cmd(Arguments.builder().shell(dockerRunCommand).build());
+    }
 
     handler.runCmds().forEach(buildBuilder::runCmd);
 

--- a/jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java
+++ b/jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java
@@ -169,4 +169,35 @@ public class WebAppGeneratorTest {
         .extracting("inline.files").asList().extracting("destName")
         .containsExactly("some-context.war");
   }
+
+  @Test
+  public void customize_withFromTargetDirConfiguredAndNoCmd_shouldNotInitializeInvalidCmdArgument() throws IOException {
+    // Given
+    final List<ImageConfiguration> originalImageConfigurations = new ArrayList<>();
+    final File buildDirectory = temporaryFolder.newFolder("build");
+    final File artifactFile = new File(buildDirectory, "artifact.war");
+    assertTrue(artifactFile.createNewFile());
+    final Properties projectProperties = new Properties();
+    projectProperties.put("jkube.generator.webapp.targetDir", "/usr/local/tomcat/webapps");
+    projectProperties.put("jkube.generator.webapp.from", "tomcat:jdk11-openjdk-slim");
+
+    when(generatorContext.getProject().getBuildDirectory()).thenReturn(buildDirectory);
+    when(generatorContext.getProject().getBuildFinalName()).thenReturn("artifact");
+    when(generatorContext.getProject().getPackaging()).thenReturn("war");
+    when(generatorContext.getProject().getVersion()).thenReturn("1.33.7-SNAPSHOT");
+    when(generatorContext.getProject().getProperties()).thenReturn(projectProperties);
+    // When
+    final List<ImageConfiguration> result = new WebAppGenerator(generatorContext)
+        .customize(originalImageConfigurations, false);
+    // Then
+    assertThat(result)
+        .isSameAs(originalImageConfigurations)
+        .hasSize(1)
+        .first()
+        .hasFieldOrPropertyWithValue("name", "%g/%a:%l")
+        .extracting(ImageConfiguration::getBuildConfiguration)
+        .hasFieldOrPropertyWithValue("from", "tomcat:jdk11-openjdk-slim")
+        .hasFieldOrPropertyWithValue("env", Collections.singletonMap("DEPLOY_DIR", "/usr/local/tomcat/webapps"))
+        .hasFieldOrPropertyWithValue("cmd", null);
+  }
 }

--- a/quickstarts/gradle/webapp-custom/build.gradle
+++ b/quickstarts/gradle/webapp-custom/build.gradle
@@ -33,7 +33,6 @@ kubernetes {
             'webapp' {
                 from = "tomcat:jdk11-openjdk-slim"
                 targetDir = "/usr/local/tomcat/webapps"
-                cmd = "catalina.sh run"
             }
         }
     }

